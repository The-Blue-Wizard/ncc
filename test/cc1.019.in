/* basic bitfield exercises */

#include "support.h"

struct s
{
    int x;
    int a : 3;
    int b : 10;
    unsigned c : 5;
    int d : 14;
};

struct s s1 = { 0, -1, 2, -3, 4 };       /* bitfield initialization */

void
prints(struct s *s)
{
    putn(s->x);
    putn(s->a);
    putn(s->b);
    putn(s->c);
    putn(s->d);
}

int
main(void)
{
    int i;

    putn(sizeof(s1));
    prints(&s1);

    for (i = 0; s1.a; ++s1.a, ++i) ;
    putn(i); 
    for (i = 0; s1.b; ++s1.b, ++i) ;
    putn(i); 
    for (i = 0; s1.c > 0; s1.c--, ++i) ;
    putn(i); 
    for (i = 0; s1.d > 0; s1.d++, ++i) ;
    putn(i); 

    s1.x = 5;
    s1.a = 100;
    s1.b = 30000;
    s1.c = -12;
    s1.d = 60000;

    prints(&s1); 
    return 0;
}

