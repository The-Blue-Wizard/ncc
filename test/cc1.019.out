8
0
-1
2
29
4
1
1022
29
8188
5
-4
304
20
-5536
